docs/tutorials/estimator_vqe.ipynb

```diff
@@ -0,0 +1,273 @@
+    "problems = []\n",
+    "while dist < total_dist:\n",
+    "    molecule = Molecule(geometry=[['H', [0., 0., 0.]], ['H', [0., 0., dist]]])\n",
+    "    driver = ElectronicStructureMoleculeDriver(molecule, basis='sto3g', \\\n",
+    "                                               driver_type=ElectronicStructureDriverType.PYSCF)\n",
+    "\n",
+    "    es_problem = ElectronicStructureProblem(driver)\n",
+    "    second_q_ops = es_problem.second_q_ops()      \n",
+    "    hamiltonian = qubit_converter.convert(second_q_ops[0], num_particles=es_problem.num_particles)\n",
+    "    sol = MinimumEigensolverResult()\n",
+    "    sol.eigenvalue = numpy_solver.compute_minimum_eigenvalue(hamiltonian).eigenvalue\n",
+    "    real_solution = es_problem.interpret(sol).total_energies[0]\n",
+    "\n",
+    "    h = hamiltonian.primitive\n",
+    "    h.coeffs = np.real(hamiltonian.coeffs)\n",
+    "    real_energies.append(real_solution)\n",
+    "    ops.append(TaperedPauliSumOp(h, hamiltonian.z2_symmetries))\n",
+    "    dists.append(dist)\n",
+    "    problems.append(es_problem)\n",
+    "    \n",
+    "    if dist > total_dist / 2:\n",
+    "        dist += incr_late\n",
+    "    else:\n",
+    "        dist += incr_early"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2baf3383",
+   "metadata": {},
+   "source": [
+    "Now we can use these hamiltonians as the observable list in our estimator factory (thus avoiding the need to create many estimator instances). The VQE routine is formalized as $\\min_\\theta \\langle \\Psi | \\hat{H} | \\Psi \\rangle $, so we just need to minimize the expectation values of the hamiltonians. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a8151031",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit.algorithms.optimizers import SPSA\n",
+    "from qiskit.circuit.library import RealAmplitudes\n",
+    "import numpy as np\n",
+    "\n",
+    "circuit = RealAmplitudes(num_qubits=2, reps=2).decompose()\n",
+    "vqe_energies = []\n",
+    "\n",
+    "with estimator_factory(\n",
+    "    circuits=[circuit],\n",
+    "    observables=ops,\n",
+    ") as estimator:\n",
+    "    for i in range(len(ops)):\n",
+    "\n",
+    "        def evaluate_expectation(x):\n",
+    "            x = list(x)\n",
+    "            results = estimator(circuit_indices=[0], observable_indices=[i], parameter_values=[x]).values[0]\n",
+    "            return np.real(results)\n",
+    "\n",
+    "\n",
+    "        initial_point = np.random.uniform(-np.pi, np.pi, 6)\n",
+    "        spsa = SPSA(100)\n",
+    "        _, vqe_solution, _ = spsa.optimize(initial_point.size, evaluate_expectation, initial_point=initial_point) \n",
+    "        \n",
+    "        sol = MinimumEigensolverResult()\n",
+    "        sol.eigenvalue = vqe_solution\n",
+    "        vqe_sol = es_problem.interpret(sol).total_energies[0]\n",
+    "        vqe_energies.append(vqe_sol)\n"
```

Comment:
This cell takes a _really_ long time to run for me, to the point that I abandoned it. I left it for a long time (at least >15 mins) and when I interrupted my `vqe_energies` was empty.

Reply:
Yea, the full run took my on the order of hours to run. I am not sure if there are ways to accelerate it (running on local simulator takes maybe 15 minutes?) but VQE is pretty time consuming.